p2p/tests/sync.rs

```diff
@@ -25,8 +25,8 @@ use p2p::{
     event::{PubSubControlEvent, SwarmEvent, SyncControlEvent},
```

Comment:
Lots of unwrap in this file. Can we change it to expect with a reasonable message that explains why it should never fail?

Reply:
These are tests and after taking a quick look most of them were about `ChannelClosed`/subsystem wrapper failure. I'm not sure if converting them to `expect()` is worth the effort.

Reply:
Apologies, I kind of missed that this is just tests.

Reply:
I still see a bunch of clippy warnings regarding `unwrap` in the `p2p/test-utils` crate. I think we can just `allow` them in that crate.

Reply:
I can fix the unwrap issue in another PR when I'm modifying those files so we don't have to make exceptions.